vms/platformvm/state/state.go

```diff
@@ -1551,38 +1551,47 @@ func (s *state) writeBlocks() error {
 	return nil
 }
 
-func (s *state) GetStatelessBlock(blockID ids.ID) (blocks.Block, choices.Status, error) {
-	if blk, ok := s.addedBlocks[blockID]; ok {
-		return blk.Blk, blk.Status, nil
+func (s *state) GetStatelessBlock(blockID ids.ID) (blocks.Block, error) {
+	if blkState, ok := s.addedBlocks[blockID]; ok {
+		if blkState.Status != choices.Accepted {
+			return nil, database.ErrNotFound
+		}
+
+		return blkState.Blk, nil
 	}
 	if blkState, ok := s.blockCache.Get(blockID); ok {
```

Comment:
Can we change the signature of:
```
	addedBlocks map[ids.ID]stateBlk // map of blockID -> Block
	// cache of blockID -> Block
	// If the block isn't known, nil is cached.
	blockCache cache.Cacher[ids.ID, *stateBlk]
```
To:
```
	addedBlocks map[ids.ID]blocks.Block // map of blockID -> Block
	// cache of blockID -> Block
	// If the block isn't known, nil is cached.
	blockCache cache.Cacher[ids.ID, blocks.Block]
```
We should be able to remove most of the status handling here if we do that.